File: python/ch03/25304.py
total = int(input())
num = int(input())

result = 0

for i in list(range(num)):
    a,b = input().split()
    a,b = int(a),int(b)
    result += a * b

if total == result:
    print('Yes')
else:
    print('No')

File: nodejs/ch03/25304.js
const fs = require("fs");
const input = fs.readFileSync("../dev/stdin").toString().trim().split("\n");

const answer = parseInt(input[0])
const count = parseInt(input[1])

let total = 0;

for (let i = 2; i < 2 + count; i++) {
  let [a, b] = input[i].split(" ");
  total += parseInt(a) * parseInt(b)
}

if (answer == total) {
  console.log("Yes")
} else {
  console.log("No")
}